test/distributed/fsdp/test_fsdp_mixed_precision.py

```diff
@@ -388,6 +402,100 @@ def test_mixed_precision_embedding_table(self):
             model.module.run_backward(loss)
             optim.step()
 
+    @skip_if_lt_x_gpu(2)
+    @skipIfNoTorchVision
+    def test_mixed_precision_resnet(self):
+        resnet_model = torchvision.models.resnet50().cuda()
+        resnet_model = nn.SyncBatchNorm.convert_sync_batchnorm(
+            copy.deepcopy(resnet_model),
+            process_group=dist.distributed_c10d._get_default_group()
+        )
+        n_bn = len(
+            [x for x in resnet_model.modules() if isinstance(x, _BatchNorm)]
+        )
+        inp = torch.ones(1, 3, 1000, 1000, device='cuda')
+        mp_config = MixedPrecision(
+            param_dtype=torch.float16,
+            reduce_dtype=torch.float16,
+            buffer_dtype=torch.float16,
+        )
+        fsdp = FSDP(
+            resnet_model,
+            auto_wrap_policy=default_auto_wrap_policy,
+            mixed_precision=mp_config
+        )
+        # Batchnorm units should be wrapped individually. Validate this by
```

Comment:
A little bit roundabout but doing this because it is tricky to get the full names of batchnorm modules due to ResNet's nested nature and find the corresponding one in FSDP